spec/requests/omniauth_callbacks_spec.rb

```diff
@@ -0,0 +1,382 @@
+        get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
+
+        user = User.find_by(email: oauth_email)
+        expect(user.token).to eq oauth_token
+      end
+
+      it 'signs in and redirect to root' do
+        get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
+
+        expect(response).to redirect_to root_path
+      end
+
+      it 'enables oauth on user' do
+        get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
+
+        user = User.find_by(email: oauth_email)
+        expect(user.oauth_enabled?).to eq true
+      end
+    end
+
+    context 'when google_oauth2 enabling fails' do
+      it 'redirects to sign in path' do
+        allow(User).to receive(:find_for_oauth).and_return(nil)
+
+        get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
+
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context 'when an invitation_token is passed in' do
+      context 'when the user logs in with the same email as the invitation' do
+        it 'sets invitation accepted at on invitee' do
+          invite_user(oauth_email)
+
+          get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
+
+          invitee = User.find_by(email: oauth_email)
+          expect(invitee.invitation_accepted_at).to be_present
+        end
+
+        it 'signs in and redirect to root' do
+          invite_user(oauth_email)
+
+          get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
+
+          expect(response).to redirect_to root_path
+        end
+
+        it 'enables oauth on user' do
+          invite_user(oauth_email)
+
+          get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
+
+          invitee = User.find_by(email: oauth_email)
+          expect(invitee.oauth_enabled?).to eq true
+        end
+      end
+
+      context 'when the user logs in with a different email from the invitation' do
+        let(:other_invitee_email) { 'no-invite@xyze.it' }
+
+        it 'does not invitation accepted at on invitee' do
+          invite_user(other_invitee_email)
+
+          get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
+
+          invitee = User.find_by(email: oauth_email)
+          expect(invitee.invitation_accepted_at).to be_blank
+        end
+
+        it 'signs in and redirect to root' do
+          invite_user(other_invitee_email)
+
+          get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
+
+          expect(response).to redirect_to root_path
+        end
+
+        it 'enables oauth on user' do
+          invite_user(other_invitee_email)
+
+          get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
+
+          user = User.find_by(email: oauth_email)
+          expect(user.oauth_enabled?).to eq true
```

Comment:
```suggestion
          invite_user(other_invitee_email)
          user = User.find_by(email: oauth_email)
          expect(user.oauth_enabled?).to be_falsey
          get omniauth_login_omniauth_callbacks_path, params: { provider: 'google' }
          expect(user.oauth_enabled?).to be_truthy
```

I think we might need an **assume** step here to make sure that the user `oauth_enabled?` status was changed unless I'm missing something?

Reply:
I used the `to change {}.from().to()` matcher here as well.. Do you think it's clear enough or having this assume step would be better?

Reply:
That works too 👍

Reply:
That works too 👍